elastic/apm-server
# fix direct agent configuration

## Motivation/summary

Manual testing with https://github.com/elastic/kibana/pull/100744 revealed
errors in the current apm-server logic. This PR introduces changes so that
apm-server correctly switches to direct agent configuration when running under
managed mode.

depends on https://github.com/elastic/kibana/pull/100744

## Checklist




## How to test these changes

1. Start elasticsearch using apm-integration-testing
2. Start a version of kibana including [this PR](https://github.com/elastic/kibana/pull/100744)
3. Start elastic-agent (not managed by fleet) with a version of `apm-server`
   including this PR
4. From the kibana UI:
  - Add apm-server as an integration to a fleet policy
  - Create an agent config in APM / Settings / Agent Config
  - Verify that the agent_config exists in the fleet policy and is nested under
    `apm-server`
5. Update the running `elastic-agent.yml` with the generated policy from the
   previous step
6. Verify that `cURL`ing the config endpoint returns your agent config
  - e.g. `curl 'http://localhost:8200/config/v1/agents?service.name=all'`
